src/deprecates.jl

```diff
@@ -48,3 +48,15 @@ findat(a::AbstractArray, b::AbstractArray) = findat!(Array{Int}(size(b)), a, b)
 @deprecate df_residual(obj::StatisticalModel) dof_residual(obj)
 
 @deprecate_binding WeightVec Weights
+
+using Base.Random: RangeGenerator
+@deprecate_binding RandIntSampler RangeGenerator
+@deprecate RangeGenerator(K::Int) RangeGenerator(1:K)
```

Comment:
OK, actually the simplest solution seems to be moving the `:RandIntSampler` code to deprecates.jl, and printing deprecation warnings from the inner constructors. No type piracy involved. The use of the binding itself does not print a message, but you can't fix the warnings without removing the uses of the type anyway, and the only package which used this unexported type was Distributions. So should be good to go.